### DISCUSSION TITLE
Switching from a zig-a-zig-ah! to a SMARTLIGHT CC2652P (+ a general question)

### DISCUSSION THREAD

**Discussion by wsw70** [Koenkk/zigbee2mqtt#8841] [General]
I have an Electrolama zig-a-zig-ah! (zzh!) dongle with all my devices connected to it.
I bought a SMARTLIGHT CC2652P dongle I would like to try on these devices.

I read _[What does and does not require repairing of all devices?](https://www.zigbee2mqtt.io/information/FAQ.html#what-does-and-does-not-require-repairing-of-all-devices)_ but I cannot find my specific case: **going from a CC2652R to a CC2652P.**

Actually, I have a hard time understanding the switch procedure to start with: on the one hand [the docs say](https://www.zigbee2mqtt.io/information/FAQ.html#requires-repairing)

>You need to re-pair all you devices when:
>
> Changing the network key (network_key), Zigbee channel (channel) or panID (pan_id) in configuration.yaml.

and on the other hand [it says](https://www.zigbee2mqtt.io/information/supported_adapters.html#notes):

> When migrating to another adapter make sure to modify your pan_id in your configuration, otherwise Zigbee2MQTT won’t start. Migrating to a different adapter may require repairing all devices.

Since I **have to change the `pan_id`**, then **I have to re-pair all my devices** (at least this is the logic I read from the docs). This logic is obviously incorrect because there are plenty of cases discussed.

- My specific question: what must be changed if i switch from a CC2652R to a CC2652P, and will it require re-pairing?
- A general question: why is the logic I get from the docs flawed?

**Comment by Hedda** [comment]
Should just need to swap them

https://www.zigbee2mqtt.io/information/FAQ.html#how-do-i-migrate-from-a-cc2531-to-a-more-powerful-coordinator-eg-zzh

backup and restore from file was also added/updated not that long ago

https://github.com/Koenkk/zigbee-herdsman/pull/395

**Reply by wsw70** [reply]
> Should just need to swap them
> 
> https://www.zigbee2mqtt.io/information/FAQ.html#how-do-i-migrate-from-a-cc2531-to-a-more-powerful-coordinator-eg-zzh

Thank you, I missed that in the documentation. 

I would be glad to do a PR with my understanding of "how to switch a dongle" to make the documentation more consistent - but it seems that the pages are maintained outside the main repo.

**Reply by Hedda** [reply]
Zigbee2MQTT.io documentation is here -> https://github.com/Koenkk/zigbee2mqtt.io

For the FAQ see -> https://github.com/Koenkk/zigbee2mqtt.io/blob/master/docs/information/FAQ.md

**Reply by wsw70** [reply]
> Zigbee2MQTT.io documentation is here -> https://github.com/Koenkk/zigbee2mqtt.io

Yes, obviously, in the place where it is expected to be. I had a look at the `Koenkk`  root and somehow missed this as well.
I should probably make some eye tests...

Thanks!

**Reply by wsw70** [reply]
> Should just need to swap them
> 
> https://www.zigbee2mqtt.io/information/FAQ.html#how-do-i-migrate-from-a-cc2531-to-a-more-powerful-coordinator-eg-zzh
> 
> backup and restore from file was also added/updated not that long ago
> 
> [Koenkk/zigbee-herdsman#395](https://github.com/Koenkk/zigbee-herdsman/pull/395)

I finally made the change a moment ago - just swapped the dongles and it worked out of the box. Thank you for the information.
